src/protego.py

```diff
@@ -0,0 +1,386 @@
+import re
+from datetime import time
+from collections import namedtuple
+import six
```

Comment:
See https://stackoverflow.com/q/20762662/939364 for recommended import sorting and grouping. https://pypi.org/project/isort/ might help.